### PR TITLE
Add check to avoid mixing usage of security_group_rule and security_group with inline rules

### PR DESCRIPTION
**Pull request by sgavinio**
## Description

Hello, this adds an AWS check to ensure that usage of security groups with inline rules and security group rule resources are not mixed.

Mixing Terraform standalone `security_group_rule` resource and `security_group` resource with inline ingress/egress rules results in rules being overwritten during Terraform apply. 

This is intended behavior as per [terraform](https://github.com/hashicorp/terraform/issues/11011#issuecomment-283076580) but can result in security group rules being overwritten if one is not careful. Big banner warning from terraform [security group docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group) :

Screen Shot 2021-08-18 at 7 58 15 AM

## Solution

This adds the check in go where we have  access to the `GetReferencedBlock` function. 

I'm not able to express how I can refer to "co-related" resources with the custom check language. Please let me know  if this is achievable, may have just missed it. Thank you 🙏 

## References

* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group
* https://github.com/hashicorp/terraform/issues/11011#issuecomment-283076580

### CODE REVIEW

File: internal/app/tfsec/rules/aws/vpc/disallow_mixed_sgr_rule.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  	security_group_id = aws_security_group.good_example_sg.id
+	type = "ingress"
+	cidr_blocks = ["10.0.0.0/16", "172.31.0.0/16"]
+}
+
+resource "aws_security_group" "good_example_sg" {
+}
+`},
+			Links: []string{
+				"https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group",
```
Comment (owenrumney) [inline internal/app/tfsec/rules/aws/vpc/disallow_mixed_sgr_rule.go:50]: Can we change this link and target the message specifically 
```suggestion
				"https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group#resource-aws_security_group",
```

**Review by owenrumney — DISMISSED** [review]
This is great, really appreciate your contribution

File: internal/app/tfsec/rules/aws/vpc/disallow_mixed_sgr_rule.go (hunk 1 above)
Comment (sgavinio) [reply internal/app/tfsec/rules/aws/vpc/disallow_mixed_sgr_rule.go]: Ack. Linked changed. Thank you.

**Review by owenrumney — APPROVED** [review]

**Comment by sgavinio** [comment]
Thanks for merging @owenrumney 🙏 . Is there a regular schedule for cutting a release? Just want to factor in when we get this from the upstream.

**Pull request merged**